### PR TITLE
feat(perf): Add new events to report modern performance metrics

### PR DESCRIPTION
**Pull request by jstoffan**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

File: src/lib/PreviewPerf.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    fcp?: number;
+    lcp?: number;
+}
+
+export default class PreviewPerf {
+    private fcpObserver: PerformanceObserver;
+
+    private lcpObserver: PerformanceObserver;
+
+    private performance: Performance = performance;
```
Comment (mxiao6) [inline src/lib/PreviewPerf.ts]: Should this be `window.performance`?

File: src/lib/Preview.js [hunk 2]
```diff
@@ -560,13 +560,10 @@ class Preview extends EventEmitter {
                 });
         }
 
-        const downloadAttemptEvent = {
+        this.emit(PREVIEW_METRIC, {
```
[the remark this answers is not in the document]
Comment (jstoffan) [reply src/lib/Preview.js]: Nice catch, thanks.

File: src/lib/PreviewPerf.ts (hunk 1 above)
Comment (jstoffan) [reply src/lib/PreviewPerf.ts]: Karma chokes when I use it that way, unfortunately. It doesn't really make a difference.

**Pull request merged**